cmd/runtimetest/main.go

```diff
@@ -191,6 +192,81 @@ func validateSysctls(spec *rspec.Spec) error {
 	return nil
 }
 
+func mountMatch(specMount rspec.Mount, sysMount rspec.Mount) error {
+	if specMount.Destination != sysMount.Destination {
+		return fmt.Errorf("mount destination expected: %v, actual: %v", specMount.Destination, sysMount.Destination)
+	}
+
+	if specMount.Type != sysMount.Type {
+		return fmt.Errorf("mount %v type expected: %v, actual: %v", specMount.Destination, specMount.Type, sysMount.Type)
+	}
+
+	if specMount.Source != sysMount.Source {
+		return fmt.Errorf("mount %v source expected: %v, actual: %v", specMount.Destination, specMount.Source, sysMount.Source)
+	}
+
+	optMap := make(map[string]bool)
+	for _, opt := range sysMount.Options {
+		optMap[opt] = true
+	}
+	for _, option := range specMount.Options {
+		if _, ok := optMap[option]; !ok {
+			return fmt.Errorf("mount %v option %v does not exist", specMount.Destination, option)
+		}
+	}
+
+	return nil
+}
+
+func containString(opts []string, str string) bool {
+	for _, opt := range opts {
+		if opt == str {
+			return true
+		}
+	}
+
+	return false
+}
+
+func validateMounts(spec *rspec.Spec) error {
+	fmt.Println("validating mounts")
+	infos, err := mount.GetMounts()
+	if err != nil {
+		return err
+	}
+
+	mountsMap := make(map[string]rspec.Mount)
+	for _, info := range infos {
+		opts := strings.Split(info.Opts, ",")
+
+		if info.Mountpoint == "/" {
+			if spec.Root.Readonly && !containString(opts, "ro") ||
+				!spec.Root.Readonly && !containString(opts, "rw") {
+				return fmt.Errorf("root readonly expected: %v, actual: %v", spec.Root.Readonly, !spec.Root.Readonly)
+			}
+		}
+
+		mountsMap[info.Mountpoint] = rspec.Mount{
+			Destination: info.Mountpoint,
+			Type:        info.Fstype,
+			Source:      info.Source,
+			Options:     append(opts, strings.Split(info.VfsOpts, ",")...)}
+
+	}
+
+	for _, specMount := range spec.Mounts {
+		if sysMount, ok := mountsMap[specMount.Destination]; ok {
+			if err := mountMatch(specMount, sysMount); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("Expected mount %v does not exist", specMount.Destination)
```

Comment:
I think we should log a warning in this case, but don't think it should be an error ([reasoning][1]).

[1]: https://github.com/opencontainers/ocitools/pull/81/files#r64780518